engine/src/flutter/impeller/renderer/backend/vulkan/render_pass_cache_unittests.cc

```diff
@@ -21,10 +21,45 @@ TEST_P(RendererTest, CachesRenderPassAndFramebuffer) {
   auto allocator = std::make_shared<RenderTargetAllocator>(
       GetContext()->GetResourceAllocator());
 
-  auto render_target =
+  RenderTarget render_target =
       allocator->CreateOffscreenMSAA(*GetContext(), {100, 100}, 1);
-  auto resolve_texture = render_target.GetColorAttachment(0).resolve_texture;
-  auto& texture_vk = TextureVK::Cast(*resolve_texture);
+  std::shared_ptr<Texture> resolve_texture =
+      render_target.GetColorAttachment(0).resolve_texture;
+  TextureVK& texture_vk = TextureVK::Cast(*resolve_texture);
+
+  EXPECT_EQ(texture_vk.GetCachedFramebuffer(), nullptr);
+  EXPECT_EQ(texture_vk.GetCachedRenderPass(), nullptr);
+
+  auto buffer = GetContext()->CreateCommandBuffer();
+  auto render_pass = buffer->CreateRenderPass(render_target);
+
+  EXPECT_NE(texture_vk.GetCachedFramebuffer(), nullptr);
+  EXPECT_NE(texture_vk.GetCachedRenderPass(), nullptr);
+
+  render_pass->EncodeCommands();
+  GetContext()->GetCommandQueue()->Submit({buffer});
```

Comment:
Assert the return result for Submit is a success.  Maybe we should have [[nodiscard]] for that.

Reply:
Done